tests/unit/backend/test_tiledb.py

```diff
@@ -0,0 +1,273 @@
+                    "ontology_term_id": "string"
+                }
+            ],
+            "tissue": [
+                {
+                    "label": "string",
+                    "ontology_term_id": "string"
+                }
+            ],
+            "explorer_url": "https://cellxgene.com/e/lolol.cxg",
+            "processing_status": {
+                "processing_status": "PENDING"
+            }
+        }
+        self.location = "tests/unit/backend/fixtures/test_tiledb/metadata"
+        TileDBData.init_db(self.location)
+        self.db = TileDBData(self.location)
+
+    def tearDown(self):
+        TileDBData.destroy_db(self.location)
+
+    def create_collection(self):
+        id = self.db.create_collection(
+            self.metadata['name'],
+            self.metadata['description'],
+            self.metadata['owner'],
+            self.metadata['contact_name'],
+            self.metadata['contact_email'],
+            self.metadata['curator_name'],
+            self.metadata['links'],
+            self.metadata['publisher_metadata']
+        )
+        return id
+
+    def test_create_collection(self):
+        id = self.create_collection()
+        coll = self.db.get_collection(id)
+        self.assertEqual(coll['name'], self.metadata['name'])
+        self.assertDictEqual(coll['publisher_metadata'], self.metadata['publisher_metadata'])
+
+    def test_edit_collection(self):
+        # Create collection
+        id = self.create_collection()
+        # Edit collection name
+        self.db.edit_collection(id, "name", "edited_name")
+        new_data = {
+                "authors": [
+                    {
+                        "family": "Agarwal",
+                        "given": "Rohan",
+                        "name": ""
+                    },
+                ],
+                "is_preprint": True,
+                "journal": "Nature",
+                "published_day": 10,
+                "published_month": 10,
+                "published_year": 2022
+            }
+        self.db.edit_collection(id, "publisher_metadata", new_data)
+        # check
+        coll = self.db.get_collection(id)
+        self.assertEqual(coll['name'], "edited_name")
+        self.assertDictEqual(coll['publisher_metadata'], new_data)
+
+    def test_add_dataset_to_collection(self):
+        # Create collection
+        id = self.create_collection()
+        # Add dataset to collection
+        dataset_id = self.db.add_dataset(id, self.dataset_metadata)
+        # Get dataset/collection
+        datasets = self.db.get_datasets(id)
+        dataset_ids = self.db.get_attribute(id, "datasets")
+        self.assertEqual(datasets[0]['cell_count'], 12)
+        self.assertEqual(dataset_ids[0], dataset_id)
+
+    def test_delete_dataset_from_collection(self):
+        # Create collection
+        id = self.create_collection()
+        # Add dataset to collection
+        did1 = self.db.add_dataset(id, self.dataset_metadata)
```

Comment:
Let's use full variable names here: `dataset_id_to_delete_1` or something like that